spark_utils/common/functions.py

```diff
@@ -70,8 +70,46 @@ def read_from_socket(socket: JobSocket, spark_session: SparkSession, **kwargs) -
     :param kwargs: Keyword arguments passed to spark read options
     :return: Spark dataframe
     """
+    if socket.data_format == 'hive':
+        return spark_session.table(socket.data_path)
+
     return spark_session \
         .read \
         .options(**kwargs) \
         .format(socket.data_format) \
         .load(socket.data_path)
+
+
+def write_to_socket(
+    data: DataFrame,
+    socket: JobSocket,
+    partition_by: List[str] = None,
+    partition_count: int = None,
+    **kwargs
```

Comment:
```suggestion
    write_options: Dict[str, str]
```
Using **kwargs in public APIs is a no-go :)